### PR TITLE
New year review falko

### PR DESCRIPTION
**Pull request by falko-strenzke**
Most changes were previously discussed in https://github.com/openpgp-pqc/draft-openpgp-pqc/issues/6. The respective comments are indicated by crossed out text in the issue.

### CODE REVIEW

File: draft-wussler-openpgp-pqc.md [hunk 1]
```diff
@@ -1321,7 +1323,7 @@ when using a sponge-based construction like KMAC. The construction using KMAC
 ensures a standardized domain separation. In this case, the processed message
 is then the concatenation of any number of key shares.
 
-More precisely, for given capacity `c` the indifferentiability proof shows that
+More precisely, for a given capacity `c` the indifferentiability proof shows that
```
Comment (fluppe2) [inline draft-wussler-openpgp-pqc.md:1336]: I think the 'a' is not necessary here.

File: draft-wussler-openpgp-pqc.md [hunk 2]
```diff
@@ -660,15 +660,17 @@ row "ECC-KEM" of {{tab-ecdh-cfrg-artifacts}}, {{tab-ecdh-nist-artifacts}}, and
 The operations `x25519Kem.encap()` and `x448Kem.encap()` are defined as
 follows:
 
- 1. Generate an ephemeral key pair {`v`, `V=vG`}, without any masking or
-    clamping as defined in [RFC7748]
+ 1. Generate an ephemeral key pair {`v`, `V=vG`} according to 
+    the procedure defined in RFC7748, that is, without any 
+    measures such as masking or clamping that are applied 
+    in previous OpenPGP specifications of this scheme.
```
Comment (fluppe2) [inline draft-wussler-openpgp-pqc.md:666]: I think the part about previous specs is not necessary here, as it is already said in §1.3.1. The formulations here should be brief. I liked the old sentence better.

File: draft-wussler-openpgp-pqc.md [hunk 3]
```diff
@@ -660,15 +660,17 @@ row "ECC-KEM" of {{tab-ecdh-cfrg-artifacts}}, {{tab-ecdh-nist-artifacts}}, and
...
 
- 2. Compute the shared point `X = vR` where `R` is the component public key
+ 3. Compute the shared point `X = vR` where `R` is the component public key
     `eccPublicKey` according to [RFC7748]
 
- 3. Set the output `eccCipherText` to `V`
+ 4. Set the output `eccCipherText` to `V`
 
- 4. Set the output `eccKeyShare` to `X` in native format
+ 5. Set the output `eccKeyShare` to `X` in native format
```
Comment (fluppe2) [inline draft-wussler-openpgp-pqc.md]: Now the list seems to go from 1 then 3,4,5. Is the 2 missing?

File: draft-wussler-openpgp-pqc.md [hunk 4]
```diff
@@ -660,8 +660,10 @@ row "ECC-KEM" of {{tab-ecdh-cfrg-artifacts}}, {{tab-ecdh-nist-artifacts}}, and
 The operations `x25519Kem.encap()` and `x448Kem.encap()` are defined as
 follows:
 
- 1. Generate an ephemeral key pair {`v`, `V=vG`}, without any masking or
-    clamping as defined in [RFC7748]
+ 1. Generate an ephemeral key pair {`v`, `V=vG`} according to
+    the procedure defined in RFC7748, that is, without any
+    measures such as masking or clamping that are applied
+    in previous OpenPGP specifications of this scheme.
```
Comment (fluppe2) [inline draft-wussler-openpgp-pqc.md:666]:
```suggestion
 1. Generate an ephemeral key pair {`v`, `V=vG`} according to
    the procedure defined in [RFC7748], that is, without any
    measures such as masking or clamping
```

File: draft-wussler-openpgp-pqc.md [hunk 5]
```diff
@@ -1052,19 +1054,29 @@ operations has to be performed:
...
- 2. Sign `dataDigest` with `eddsa.sign()` from {{eddsa-signature}}
+ 2. Create the EdDSA signature over `dataDigest` with `eddsa.sign()` from
+    {{eddsa-signature}}
 
- 3. Sign `dataDigest` with `dilithium.sign()` from {{dilithium-signature}}
+ 3. Create the Dilithium signature over `dataDigest` with `dilithium.sign()`
+    from {{dilithium-signature}}
+
+ 4. Encode the EdDSA and Dilithium signatures according to the the packet
+    structure given in {{ecc-dilithium-sig-packet}}.
```
Comment (fluppe2) [inline draft-wussler-openpgp-pqc.md:1064]:
```suggestion
 4. Encode the EdDSA and Dilithium signatures according to the packet
    structure given in {{ecc-dilithium-sig-packet}}.
```

File: draft-wussler-openpgp-pqc.md [hunk 6]
```diff
@@ -1052,19 +1054,29 @@ operations has to be performed:
...
- 2. Sign `dataDigest` with `ecdsa.sign()` from {{ecdsa-signature}}
+ 2. Create the ECDSA signature over `dataDigest` with `ecdsa.sign()` from
+    {{ecdsa-signature}}
+
+ 3. Create the Dilithium signature over `dataDigest` with `dilithium.sign()`
+    from {{dilithium-signature}}
 
- 3. Sign `dataDigest` with `dilithium.sign()` from {{dilithium-signature}}
+ 4. Encode the ECDSA and Dilithium signatures according to the the packet
+    structure given in {{ecc-dilithium-sig-packet}}.
```
Comment (fluppe2) [inline draft-wussler-openpgp-pqc.md:1079]:
```suggestion
 4. Encode the ECDSA and Dilithium signatures according to the packet
    structure given in {{ecc-dilithium-sig-packet}}.
```

File: draft-wussler-openpgp-pqc.md [hunk 7]
```diff
@@ -1364,11 +1376,11 @@ signatures already provide a salted hash of the appropriate size.
...
-In order not to extend the attack surface, we bind the hash algorithm to the
-hash internally used into the signature algorithm. Dilithium internally uses a
-SHAKE256 digest, therefore we require SHA3 in the Dilithium + ECC signature
-packet. In the case of SPHINCS+ the internal hash algorithm varies based on
-the algorithm and parameter ID.
+In order not to extend the attack surface, we bind the hash algorithm used for
+message digestion to the hash algorithm used internally by the signature
+algorithm.  Dilithium internally uses a SHAKE256 digest, therefore we require
+SHA3 in the Dilithium + ECC signature packet. In the case of SPHINCS+ the
+internal hash algorithm varies based on the algorithm and parameter ID.
```
Comment (fluppe2) [inline draft-wussler-openpgp-pqc.md:1383]:
```suggestion
In order not to extend the attack surface, we bind the hash algorithm used for
message digestion to the hash algorithm used internally by the signature
algorithm. Dilithium internally uses a SHAKE256 digest, therefore we require
SHA3 in the Dilithium + ECC signature packet. In the case of SPHINCS+ the
internal hash algorithm varies based on the algorithm and parameter ID.
```

File: draft-wussler-openpgp-pqc.md (hunk 4 above)
Comment (wussler) [reply draft-wussler-openpgp-pqc.md:666]: Probably it's better to remove the reference to older OpenPGP specs, but i like the sentence either way

File: draft-wussler-openpgp-pqc.md (hunk 1 above)
Comment (wussler) [reply draft-wussler-openpgp-pqc.md:1336]: Also to me it "sounds better" with the `a` but I see Stavros' point

**Review by wussler — APPROVED** [review]

File: draft-wussler-openpgp-pqc.md (hunk 3 above)
Comment (wussler) [reply draft-wussler-openpgp-pqc.md]: FYI this does not matter with markdown, but I agree it's nice to keep tidy

File: draft-wussler-openpgp-pqc.md (hunk 1 above)
Comment (falko-strenzke) [reply draft-wussler-openpgp-pqc.md:1336]:
```suggestion
More precisely, for given capacity `c` the indifferentiability proof shows that
```
I don't know english grammar well enough to be sure but also I don't mind too much. Maybe we get a review from an english native speaker at some point.

Comment (fluppe2) [reply draft-wussler-openpgp-pqc.md:1335]: For me introducing an 'a' is fine.

**Review by fluppe2 — APPROVED** [review]

**Review by wussler — APPROVED** [review]

**Pull request merged**